lib/json_diff_ex.ex

```diff
@@ -131,24 +135,31 @@ defmodule JsonDiffEx do
   end
 
   @spec do_diff(binary | integer | float, binary | integer | float) :: map | nil
-  defp do_diff(i1, i2) when not (is_list(i1) and is_list(i2))
+  defp do_diff(i1, i2, opts) when not (is_list(i1) and is_list(i2))
                     and not (is_map(i1) and is_map(i2)) do
-    case i1 === i2 do
-      true -> nil
-      false -> [i1, i2]
+    if Keyword.get(opts, :strict_equality, @default_strict_equality) do
```

Comment:
I would do something like
```
compare = if Keyword.get(opts, :strict_equality, @default_strict_equality) do
  &===/2
else
  &==/2
end
case compare.(i1, i2) do ...
```

That way you cut down on the duplication in the case body.